bagelbids/cli.py

```diff
@@ -193,32 +221,38 @@ def bids(
     for bids_sub_id in layout.get_subjects():
         pheno_subject = pheno_subject_dict.get(f"sub-{bids_sub_id}")
         session_list = []
+
+        bids_sessions = layout.get_sessions(subject=bids_sub_id)
+        if not bids_sessions and layout.get_datatypes(subject=bids_sub_id):
+            bids_sessions = [None]
+
         # For some reason .get_sessions() doesn't always follow alphanumeric order
         # By default (without sorting) the session lists look like ["02", "01"] per subject
-        for session in sorted(layout.get_sessions(subject=bids_sub_id)):
-            image_list = []
-            for bids_file in layout.get(
-                subject=bids_sub_id,
+        for session in sorted(bids_sessions):
+            image_list = create_acquisitions(
+                layout=layout,
+                bids_sub_id=bids_sub_id,
                 session=session,
-                extension=[".nii", ".nii.gz"],
-            ):
-                # If the suffix of a BIDS file is not recognized, then ignore
-                mapped_term = map_term_to_namespace(
-                    bids_file.get_entities().get("suffix"),
-                    namespace=mappings.BIDS,
-                )
-                if mapped_term:
-                    image_list.append(
-                        models.Acquisition(
-                            hasContrastType=models.Image(
-                                identifier=mapped_term
-                            )
-                        )
-                    )
+            )
+
+            if not image_list:
+                continue
+
+            # TODO: Currently if a subject has BIDS data but no "ses-" directories (e.g., only 1 session),
+            # we create a session for that subject with a custom label "ses-nb01" to be added to the graph
+            # so the API can still find the session-level information.
+            # This should be revisited in the future as for these cases the resulting dataset object is not
+            # an exact representation of what's on disk.
+            session_label = "nb01" if bids_sessions == [None] else session
```

Comment:
that doesn't look quite right - although it should come out to the same thing. if the `session` is `None`, then we don't use it for the name but instead use the `"nb01"` string. so it should be `session is None` instead.